docs/source/cli-reference/core-options.rst

```diff
@@ -119,3 +119,28 @@ Comparing Progress Message Options
     An example usage::
 
         scancode -clieu --json-pp sample.json samples --max-in-memory -1
+
+----
+
+``--max_depth`` Option
+----------------------------------
+
+    Normally, the scan takes place upto the maximum level of nesting of directories possible. But
+    using the ``--max-depth`` option, the depth of the scan can be limited. This can reduce the time
+    taken for the scan when deeper directories are not relevant.
+
+    Note that the ``--max-depth`` option will be ignored if you are scanning from a JSON file using
+    the ``--from-json`` option. In that case, the original depth is used.
+
+    Some INTEGER values of the ``--max-depth INTEGER`` option:
+
+    - **4**     - Scan only upto 4 levels of subdirectories
+    - **0**     - Default, Scan upto maximum levels of subdirectories (no depth limit)
+    - **-1**    - Same effect as when set to 0
```

Comment:
Let's not keep option that have no effect around.

Reply:
So line `138` and `139` both need to be removed. Is that right?

Reply:
Just the negative value... since this is not supported this is not needed. Also the ddefault may not need its own line of doc, so yes  138 and 139 can be removed

Reply:
I have removed these and added a usage example in this section.